home/api_v3.py

```diff
@@ -103,74 +104,58 @@ class ContentPagesV3APIViewset(PagesAPIViewSet):
             "tag",
             "trigger",
             "page",
-            "qa",
-            "whatsapp",
-            "viber",
-            "messenger",
-            "web",
-            "s",
-            "sms",
-            "ussd",
+            "return_drafts",
+            "channel",
+            "slug",
         ]
     )
 
     pagination_class = PageNumberPagination
 
     def detail_view(self, request, pk=None, slug=None):
+        if "channel" in self.request.query_params:
+            channel = self.request.query_params.get("channel", "").lower()
+            if channel not in {"web", "whatsapp", "sms", "ussd", "messenger", "viber"}:
+                raise ValidationError(
+                    {"channel": [f"Channel matching query '{channel}' does not exist."]}
+                )
         if slug is not None:
             self.lookup_field = "slug"
+
         else:
             self.lookup_field = "pk"
 
         try:
-            if "qa" in request.GET and request.GET["qa"].lower() == "true":
+            if (
+                "return_drafts" in request.GET
+                and self.request.query_params.get("return_drafts", "").lower() == "true"
```

Comment:
I don't think you need the `"return_drafts" in request.GET` anymore, because `query_params.get("return_drafts", "")` takes care of that.